src/deriv_api/DerivAPIBasic.js

```diff
@@ -185,11 +191,23 @@ export default class DerivAPIBasic extends DerivAPICalls {
         return this.subscription_manager.forgetAll(...types);
     }
 
+    keepAlivePing() {
+        this.ping({ ping: 1 });
+        this.reconnect_timeout = setTimeout(this.reconnect.bind(this), 5000);
+    }
+
+    pong() {
+        clearTimeout(this.reconnect_timeout);
+    }
+
     openHandler() {
         this.events.next({
             name: 'open',
         });
-
+        if (this.shouldReconnect) {
+            this.pong(); // clear previous timeouts
+            setInterval(this.keepAlivePing.bind(this), 30000);
```

Comment:
can we clear this interval when the connection is closed?

Reply:
Thanks @balakrishna-binary for pointing this. Added clearInterval before reconnect attempt